src/Sylius/Bundle/ProductBundle/Form/EventSubscriber/BuildAttributesFormSubscriber.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ProductBundle\Form\EventSubscriber;
+
+use Sylius\Component\Product\Model\ProductAttributeInterface;
+use Sylius\Component\Product\Model\ProductAttributeValueInterface;
+use Sylius\Component\Product\Model\ProductInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Resource\Translation\Provider\TranslationLocaleProviderInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+final class BuildAttributesFormSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var FactoryInterface
+     */
+    private $attributeValueFactory;
+
+    /**
+     * @var TranslationLocaleProviderInterface
+     */
+    private $localeProvider;
+
+    /**
+     * @param FactoryInterface $attributeValueFactory
+     * @param TranslationLocaleProviderInterface $localeProvider
+     */
+    public function __construct(
+        FactoryInterface $attributeValueFactory,
+        TranslationLocaleProviderInterface $localeProvider
+    ) {
+        $this->attributeValueFactory = $attributeValueFactory;
+        $this->localeProvider = $localeProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            FormEvents::PRE_SET_DATA => 'preSetData',
+        ];
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function preSetData(FormEvent $event)
+    {
+        /** @var ProductInterface $product */
+        $product = $event->getData();
+
+        $localeCodes = $this->localeProvider->getDefinedLocalesCodes();
+        $defaultLocaleCode = $this->localeProvider->getDefaultLocaleCode();
+
+        $attributes = $product->getAttributes()->filter(
+            function (ProductAttributeValueInterface $attribute) use ($defaultLocaleCode) {
+                return $attribute->getLocaleCode() === $defaultLocaleCode;
+            }
+        );
+
+        foreach ($attributes as $attribute) {
+            foreach ($localeCodes as $localeCode) {
```

Comment:
```php
    /**
     * @param FormEvent $event
     */
    public function preSetData(FormEvent $event)
    {
        /** @var ProductInterface $product */
        $product = $event->getData();

        $localeCodes = $this->localeProvider->getDefinedLocalesCodes();
        $defaultLocaleCode = $this->localeProvider->getDefaultLocaleCode();

        $attributes = $product->getAttributes()->filter(
            function (ProductAttributeValueInterface $attribute) use ($defaultLocaleCode) {
                return $attribute->getLocaleCode() === $defaultLocaleCode;
            }
        );

        foreach ($attributes as $attribute) {
            $this->resolveLocalizedAttributes($localeCodes, $product, $attribute);
        }
    }

    /**
     * @param array $localeCodes
     * @param ProductInterface $product
     * @param ProductAttributeInterface $productAttribute
     */
    private function resolveLocalizedAttributes(array $localeCodes, ProductInterface $product, ProductAttributeInterface $productAttribute)
    {
        foreach ($localeCodes as $localeCode) {
            if (!$product->hasAttributeByCodeAndLocale($productAttribute->getCode(), $localeCode)) {
                $attributeValue = $this->createProductAttributeValue($productAttribute->getAttribute(), $localeCode);
                $product->addAttribute($attributeValue);
            }
        }
    }
```
or
```php

    /**
     * @param FormEvent $event
     */
    public function preSetData(FormEvent $event)
    {
        /** @var ProductInterface $product */
        $product = $event->getData();

        $defaultLocaleCode = $this->localeProvider->getDefaultLocaleCode();

        $attributes = $product->getAttributes()->filter(
            function (ProductAttributeValueInterface $attribute) use ($defaultLocaleCode) {
                return $attribute->getLocaleCode() === $defaultLocaleCode;
            }
        );

        foreach ($attributes as $attribute) {
            $this->resolveLocalizedAttributes($product, $attribute);
        }
    }

    /**
     * @param ProductInterface $product
     * @param ProductAttributeInterface $productAttribute
     */
    private function resolveLocalizedAttributes(ProductInterface $product, ProductAttributeInterface $productAttribute)
    {
        $localeCodes = $this->localeProvider->getDefinedLocalesCodes();

        foreach ($localeCodes as $localeCode) {
            if (!$product->hasAttributeByCodeAndLocale($productAttribute->getCode(), $localeCode)) {
                $attributeValue = $this->createProductAttributeValue($productAttribute->getAttribute(), $localeCode);
                $product->addAttribute($attributeValue);
            }
        }
    }
```

Anyway, we should add `Assert::isInstanceOf(ProductInterface)` just after `$product = $event->getData();`